apps/prairielearn/src/models/grading-job.ts

```diff
@@ -10,3 +14,13 @@ export function gradingJobStatus(gradingJob: GradingJob | null): GradingJobStatu
   if (gradingJob.grading_submitted_at != null) return 'queued';
   return 'requested';
 }
+
+/**
+ * Select a grading job by ID, returning null if it does not exist.
+ *
+ * @param grading_job_id The grading job ID.
+ * @returns The grading job, or null if it does not exist.
+ */
+export async function selectOptionalGradingJob(grading_job_id: string): Promise<GradingJob | null> {
+  return queryOptionalRow(sql.select_grading_job, { grading_job_id }, GradingJobSchema);
```

Comment:
Nit (I think we want to do this? Might need Prettier):

```suggestion
  return await queryOptionalRow(sql.select_grading_job, { grading_job_id }, GradingJobSchema);
```

Reply:
Yes! This is what happens when I let copilot write functions for me 😅